boot/CMakeLists.txt

```diff
@@ -15,10 +15,12 @@ else()
     message(FATAL_ERROR "Unknown ARCH '" ${ARCH} "', cannot generate a valid UEFI boot filename.")
 endif()
 
-add_custom_target(efisys
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/efisys.bin
     COMMAND native-fatten ${CMAKE_CURRENT_BINARY_DIR}/efisys.bin -format 2880 EFIBOOT -boot ${CMAKE_CURRENT_BINARY_DIR}/freeldr/bootsect/fat.bin -mkdir EFI -mkdir EFI/BOOT -add $<TARGET_FILE:bootmgfw> EFI/BOOT/boot${EFI_PLATFORM_ID}.efi
     DEPENDS native-fatten fat bootmgfw
     VERBATIM)
+add_custom_target(efisys DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/efisys.bin)
```

Comment:
Why isn't CMake clever enough to understand that the former (existing) code should have the same behaviour as the new one?

Reply:
It doesn't. add_custom_target(COMMAND) creates an always out-of-date target, since you don't provide info about what the command generates, whereas add_custom_target(DEPENDS) properly creates a target which is as up-to-date as the command output.

Same goes for the cdrom targets.